imports/ui/components/Paginate.jsx

```diff
@@ -0,0 +1,55 @@
+import { Center, SimpleGrid, Wrap } from '@chakra-ui/react';
+import React, { useEffect, useState } from 'react';
+import ReactPaginate from 'react-paginate';
+
+import '../utils/styles/paginate.css';
+
+const defaultItemsPerPage = 12;
+
+function PaginatedItems({ items, itemsPerPage = defaultItemsPerPage, grid, wrap, children }) {
+  const [currentItems, setCurrentItems] = useState(null);
+  const [pageCount, setPageCount] = useState(0);
+  const [itemOffset, setItemOffset] = useState(0);
+
+  useEffect(() => {
+    const endOffset = itemOffset + itemsPerPage;
+    setCurrentItems(items.slice(itemOffset, endOffset));
+    setPageCount(Math.ceil(items.length / itemsPerPage));
+  }, [itemOffset, itemsPerPage]);
+
+  const handlePageClick = (event) => {
+    const newOffset = (event.selected * itemsPerPage) % items.length;
+    setItemOffset(newOffset);
+  };
+
+  return (
+    <>
+      {grid && (
+        <SimpleGrid {...grid}>
```

Comment:
adding a length check `processesRendered.length > 0` fixed the problem. what was buggy was about an empty array.

Reply:
maybe instead of grid and wrap, we can make one prop container and then check if grid or wrap under that ??

Reply:
> adding a length check `processesRendered.length > 0` fixed the problem. what was buggy was about an empty array.

The problem was because `useEffect` hook within the `Paginate` component needed the `items` as dependency so that it could update.

I pushed it now